libmetal/lib/system/zephyr/sleep.h

```diff
@@ -28,6 +28,11 @@ static inline int __metal_sleep_usec(unsigned int usec)
 	return 0;
 }
 
+static inline void __metal_cpu_yield(void)
```

Comment:
proposal to replace by `metal_wait_usec() `to put in `zephyr/sys.h`
```c
static inline void metal_wait_usec(void) {
	k_busy_wait(usec)
}
```
No need here to use Z_SPIN_delay, with that we can support the hosted on different system